### DISCUSSION TITLE
Table Extraction Adds 0 to cell

### DISCUSSION THREAD

**Discussion by flickp** [jsvine/pdfplumber#1203] [Ask for help with specific PDFs]
I have a pdf that can be found [here](https://www.cs.ny.gov/pio/WorkersCompAnnualReport2011.pdf) that when I extract a table on page 16, an extra 0 is added to a cell in the last row of the table.  In the pdf there is no 0 at this location.

`pdf_file = "PDFs\\WorkersCompAnnualReport2011.pdf"`
`pdf = pdfplumber.open(pdf_file)`
`page = pdf.pages[16]`
`im = page.to_image()`
`im.debug_tablefinder()`

![Output](https://github.com/user-attachments/assets/169c71d3-06ee-4ec3-8f24-7f72a612528b)

When I extract the table, there is an extra column with a 0 in it that is not present in the table:

`# Extract the table from the current page`
`# snap_tolerance needed to combine the newline-separated years in the header`
`table = page.extract_table(table_settings={'snap_tolerance': 6})`
`table[32]`

<img width="448" alt="Screenshot 2024-09-12 090309" src="https://github.com/user-attachments/assets/aef74711-7320-4523-be40-5d9809694bc5">

Any advice on what may be happening?
Thanks

**Comment by jsvine** [comment]
Interesting! Doing this:

```python
im = page.to_image(resolution=150)
im.reset().draw_rects([ c for c in page.chars if c["text"] == "0" ])
```

... produces this:

![image](https://github.com/user-attachments/assets/6a3104f2-5c0f-4ee6-849f-032ecb41b77f)

Looking toward the bottom of the page, there *does* seem to be an extra `0` on the page that isn't otherwise visible. Characters can appear non-visible for various reasons, such as being the same color as the background (not the issue in this case), being overdrawn by another graphical element, or masking.

**Comment by flickp** [comment]
Ah, did not see that!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
